namesys/namesys.go

```diff
@@ -1,59 +1,95 @@
 package namesys
 
 import (
+	"strings"
+
 	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
 	ci "github.com/ipfs/go-ipfs/p2p/crypto"
 	path "github.com/ipfs/go-ipfs/path"
 	routing "github.com/ipfs/go-ipfs/routing"
 )
 
-// ipnsNameSystem implements IPNS naming.
+// ipfs NameSystem implements generic IPFS naming.
 //
-// Uses three Resolvers:
+// Uses several Resolvers:
 // (a) ipfs routing naming: SFS-like PKI names.
 // (b) dns domains: resolves using links in DNS TXT records
 // (c) proquints: interprets string as the raw byte data.
 //
 // It can only publish to: (a) ipfs routing naming.
 //
-type ipns struct {
-	resolvers []Resolver
-	publisher Publisher
+type ipfs struct {
+	resolvers map[string]Resolver
+	publishers map[string]Publisher
 }
 
 // NewNameSystem will construct the IPFS naming system based on Routing
 func NewNameSystem(r routing.IpfsRouting) NameSystem {
-	return &ipns{
-		resolvers: []Resolver{
-			new(DNSResolver),
-			new(ProquintResolver),
-			NewRoutingResolver(r),
+	return &ipfs{
+		resolvers: map[string]Resolver{
+			"/dnslink/": new(DNSResolver),
+			"/proquint/": new(ProquintResolver),
```

Comment:
:-1: let's not make proquints a protocol-- as we both mentioned above, proquint is more of an encoding. so maybe we should be prefixing it (if the text doesnt follow an _always predictable_ pattern)